dio/lib/src/transformers/sync_transformer.dart

```diff
@@ -125,7 +125,7 @@ class SyncTransformer extends Transformer {
     );
     final String? response;
     if (options.responseDecoder != null) {
-      response = options.responseDecoder!(
+      response = await options.responseDecoder!(
```

Comment:
This should not await the return value if it's not a future. The code should then roughly look like the following code:
```dart
      final responseValue = await options.responseDecoder!(...);
      if (responseValue is Future) {
        response = await responseValue;
      } else {
        response = responseValue;
      }
```